### PR TITLE
Register services with Consul

### PR DESCRIPTION
**Pull request by diptanu**

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -136,6 +146,9 @@ func NewClient(cfg *config.Config) (*Client, error) {
 
 	// Start the client!
 	go c.run()
+
+	// Start the consul client
+	go c.consulClient.SyncWithConsul()
```
Comment (cbednarski) [inline client/client.go]: Do we need synchronization around this? `go` statements without a channel or mutex are a code smell. For example it's possible for this function to get to `return c, nil` before the `SyncWithConsul` call is made. Does that matter?

Comment (diptanu) [inline client/client.go]: In this case it doesn't matter, as consul sync will run asynchronously and the client need not wait.

File: client/consul.go [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+				}
+			}
+
+			// See if consul thinks we have some services which are not running
+			// anymore on the node. We de-register those services
+			for serviceId := range consulServices {
+				if serviceId == "consul" {
+					continue
+				}
+				if _, ok := c.trackedServices[serviceId]; !ok {
```
Comment (cbednarski) [inline client/consul.go]: It looks like this clobbers other services that may be pre-installed on the host (e.g. monitoring or logging tools installed via puppet or baked into an AMI).

Does `trackedServices` get serialized anywhere?

Comment (diptanu) [inline client/consul.go]: We will document this behavior. We are assuming the cluster manager controls all the externally discoverable services on a machine.

**Pull request merged**